packages/react-signature-frame/src/index.js

```diff
@@ -17,9 +17,42 @@ class AnvilSignatureFrame extends React.Component {
     window.removeEventListener('message', this.handleSignFinish)
   }
 
-  handleSignFinish = ({ origin, data: url }) => {
+  handleSignFinish = ({ origin, data }) => {
     if (this.props.anvilURL !== origin) return
-    this.props.onFinish(url)
+    if (typeof data === 'string') {
+      this.props.onFinish(data)
+
+      // parse query params into an object
+      const searchStr = data.split('?')[1]
+      let payload, searchObj
+      if (typeof URLSearchParams !== 'undefined') {
+        searchObj = new URLSearchParams(searchStr)
+        payload = {
+          action: 'signerComplete',
+          signerStatus: searchObj.get('signerStatus'),
+          signerEid: searchObj.get('signerEid'),
+          nextSignerEid: searchObj.get('nextSignerEid'),
+          documentGroupStatus: searchObj.get('documentGroupStatus'),
+          documentGroupEid: searchObj.get('documentGroupEid'),
+          etchPacketEid: searchObj.get('etchPacketEid'),
+          weldDataEid: searchObj.get('weldDataEid'),
+        }
+      } else {
+        searchObj = JSON.parse('{"' + decodeURI(searchStr).replace(/"/g, '\\"').replace(/&/g, '","').replace(/=/g, '":"') + '"}')
+        payload = {
+          action: 'signerComplete',
+          signerStatus: searchObj.signerStatus ?? null,
+          signerEid: searchObj.signerEid ?? null,
+          nextSignerEid: searchObj.nextSignerEid ?? null,
+          documentGroupStatus: searchObj.documentGroupStatus ?? null,
+          documentGroupEid: searchObj.documentGroupEid ?? null,
+          etchPacketEid: searchObj.etchPacketEid ?? null,
+          weldDataEid: searchObj.weldDataEid ?? null,
+        }
+      }
+
+      if (!searchObj.error && !searchObj.errorType) this.props.onFinishSigning(payload)
```

Comment:
Does this work in the URLSearchParams case? Or would it need to call `.get()`? Would it make more sense to put those things into the payload?

Reply:
Yeah, would need to call `.get()`. Good catch.